### ISSUE TITLE
Dependency change in spatial _plotutils.py

### ISSUE THREAD

**Issue by Mdlkxzmcp** [scipy/scipy#7199]
The ax.hold() and ax.ishold() methods are [no longer supported.](http://matplotlib.org/api/api_changes.html#hold-functionality-deprecated)  This doesn't break the functions within, but it raises warnings:

> /usr/local/lib/python3.5/dist-packages/scipy/spatial/_plotutils.py:17: MatplotlibDeprecationWarning: The ishold function was deprecated in version 2.0.
>   was_held = ax.ishold()
> 
> /usr/local/lib/python3.5/dist-packages/scipy/spatial/_plotutils.py:19: MatplotlibDeprecationWarning: axes.hold is deprecated.
>     See the API Changes document (http://matplotlib.org/api/api_changes.html)
>     for more details.
>   ax.hold(True)
> 
> /usr/local/lib/python3.5/dist-packages/scipy/spatial/_plotutils.py:22: MatplotlibDeprecationWarning: axes.hold is deprecated.
>     See the API Changes document (http://matplotlib.org/api/api_changes.html)
>     for more details.
>   ax.hold(was_held)`

**Comment by perimosocordiae** [comment]
I started addressing this in gh-6994, so newer versions will at least produce fewer deprecation warnings. Unfortunately we can't fully remove the deprecated calls until we stop supporting matplotlib 1.x versions, so we're stuck with the warnings for now.

**Comment by pv** [comment]
The code could check `matplotlib.__version__` and act accordingly.

**Comment by balzer82** [comment]
This now breaks the function.

```python
/anaconda3/lib/python3.6/site-packages/scipy/spatial/_plotutils.py in _held_figure(func, obj, ax, **kw)
     18     # As of matplotlib 2.0, the "hold" mechanism is deprecated.
     19     # When matplotlib 1.x is no longer supported, this check can be removed.
---> 20     was_held = ax.ishold()
     21     if was_held:
     22         return func(obj, ax=ax, **kw)

AttributeError: 'AxesSubplot' object has no attribute 'ishold'
```

**Comment by perimosocordiae** [comment]
matplotlib is an optional dependency of scipy, so maybe it's time to drop support for older versions? (That is, 1.x versions.)

If not, we'll need another workaround here for matplotlib v3 support.

**Comment by rgommers** [comment]
> This now breaks the function. (matplotlib '3.0.2')

This was fixed in commit d6e321ab. Current status seems fine, and when we do drop 1.x there's a little cleanup we could do.

Closing, no need to keep an issue for a potential future cleanup open.

Thanks all.
